tests/client/amo/containers/TestApp.js

```diff
@@ -28,4 +30,29 @@ describe('App', () => {
     assert.equal(rootNode.tagName.toLowerCase(), 'div');
     assert.equal(rootNode.querySelector('p').textContent, 'The component');
   });
+
+  it('shows a log in button', () => {
+    const i18n = getFakeI18nInst();
+    const handleLogIn = sinon.spy();
+    const root = renderIntoDocument(
+      <AppBase i18n={i18n} isAuthenticated={false} handleLogIn={handleLogIn} />
+    );
+    const button = findDOMNode(root).querySelector('footer button');
+    assert.equal(button.textContent, 'Log in');
+    Simulate.click(button);
+    assert.ok(handleLogIn.called);
+  });
+
+  it('tells you if you are logged in', () => {
+    const i18n = getFakeI18nInst();
+    const root = renderIntoDocument(<AppBase i18n={i18n} isAuthenticated />);
+    assert.equal(findDOMNode(root).querySelector('footer').textContent, 'You are logged in');
+  });
+
+  it('updates the location on handleLogIn', () => {
+    const _window = { location: '/' };
```

Comment:
since `/` is one of the possibilities for mis-parsing the state, maybe set this to `{ location: '/elsewhere' }`